app/src/pages/common/modals/confirmModal.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { hideModalAction, confirmModalAction } from 'controllers/modal';
+
+@withModal('confirmModal')
+@connect(null, {
+  confirmModal: confirmModalAction,
+  hideModal: hideModalAction,
+})
+export class ConfirmModal extends Component {
+  static propTypes = {
+    data: PropTypes.object,
+    confirmModal: PropTypes.func.isRequired,
+    hideModal: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    data: {},
+  };
+
+  render() {
+    const { message, onConfirm, title, confirmText, cancelText } = this.props.data;
+    const { confirmModal, hideModal } = this.props;
+    return (
+      <ModalLayout
+        title={title}
+        okButton={{
+          text: confirmText,
+          onClick: (closeModal) => {
+            confirmModal();
+            closeModal();
+            onConfirm();
+            hideModal();
```

Comment:
Is it necessary to use `hideModal` after `closeModal`? As far as I see it is called automatically after closing a modal.

Reply:
`hideModal` is called in `onExited` event of `CSSTransition` component in `modalLayout`. Unfortunately, it doesn't work properly when you perform an asynchronous redirect to another page.

Reply:
Maybe dispatching `hideModalAction` in saga would look less confusing.

Reply:
I suggest dispatch it in confirmModalAction. Would it be OK?

Reply:
You can't dispatch two actions in one (there's a task to remove all thunk usage in favor of saga - no need to keep both libraries).
Somewhere [here](https://github.com/reportportal/service-ui/pull/1317/files#diff-4b647578726008404857fb737592a00cR11), I think:
```
if (confirmed) {
  yield put(hideModalAction());
}
```